execution-core/src/transfer.rs

```diff
@@ -93,6 +94,30 @@ impl Serializable<{ StealthAddress::SIZE + u64::SIZE + BlsPublicKey::SIZE }>
     }
 }
 
+/// Data for either contract call or contract deployment.
+#[derive(Debug, Clone, PartialEq, Eq, Archive, Serialize, Deserialize)]
+#[archive_attr(derive(CheckBytes))]
+pub enum CallOrDeploy {
+    /// Data for a contract call.
+    Call(ContractCall),
+    /// Data for a contract deployment.
+    Deploy(ContractDeploy),
+}
+
+/// Data for performing a contract deployment
+#[derive(Debug, Clone, PartialEq, Eq, Archive, Serialize, Deserialize)]
+#[archive_attr(derive(CheckBytes))]
+pub struct ContractDeploy {
+    /// The optional ID of the contract to be deployed.
+    pub contract_id: Option<ContractId>,
```

Comment:
A caller should never be able to determine the contract ID. As such I don't think the contract ID should be included here
```suggestion
```

Reply:
If that is so, I can remove it.

Reply:
Yeah, it should be a network decision. For vanity addresses, the nonce should be used.

Reply:
OK then, I will remove it.